File: ALODWebUtility.CS/Printing/PrintingUtil.cs
using System;
using System.Collections.Generic;
using ALOD.Core.Domain.Modules.Lod;
using ALOD.Core.Domain.Workflow;
using ALOD.Core.Utils;
using ALOD.Data.Services;

namespace ALODWebUtility.PrintingUtil
{
    public static class PrintingUtil
    {
        public static bool CheckForPriorStatus(short status, int refId)
        {
            LineOfDuty lod = LodService.GetById(refId);

            if (status == LodStatusCode.Complete)
            {
                IList<WorkStatusTracking> ws = WorkFlowService.GetWorkStatusTracking(lod.Id, ModuleType.LOD);

                if (ws != null)
                {
                    if (ws.Count > 1)
                    {
                        if (ws[1].WorkflowStatus.StatusCodeType.Id == LodStatusCode.AppointingAutorityReview)
                        {
                            return true;
                        }
                        else
                        {
                            return false;
                        }
                    }
                }
            }

            return false;
        }

        public static void ClearFormField(Printing.PDFForm doc, string field)
        {
            SetFormField(doc, field, string.Empty);
        }

        public static string GetFindingFormText(Finding findingValue)
        {
            switch (findingValue)
            {
                case Finding.In_Line_Of_Duty:
                    return "Line of Duty";
                case Finding.Epts_Lod_Not_Applicable:
                    return "EPTS-LOD Not Applicable";
                case Finding.Nlod_Due_To_Own_Misconduct:
                    return "Not ILOD-Due to Own Misconduct";
                case Finding.Epts_Service_Aggravated:
                    return "EPTS-Service Aggravated";
                case Finding.Nlod_Not_Due_To_OwnMisconduct:
                    return "Not ILOD-Not Due to Own Misconduct";
                case Finding.Recommend_Formal_Investigation:
                    return "Formal Investigation";
                default:
                    return string.Empty;
            }
        }

        public static bool IsValidSignature(SignatureEntry signature)
        {
            if (signature == null)
            {
                return false;
            }

            return signature.IsSigned;
        }

        public static string RemoveNewLinesFromString(string data)
        {
            if (string.IsNullOrEmpty(data))
            {
                return string.Empty;
            }

            string replacedString = data;
            bool replaceFinished = false;
            int len;

            while (!replaceFinished)
            {
                len = replacedString.Length;
                replacedString = replacedString.Replace(Environment.NewLine, " ");

                if (len == replacedString.Length)
                {
                    replaceFinished = true;
                }
            }

            return replacedString;
        }

        public static void SetCheckboxField(Printing.PDFForm doc, string field, bool? value)
        {
            if (!value.HasValue || value.Value == false)
            {
                return;
            }

            SetFormField(doc, field, "1");
        }

        public static void SetDateTimeField(Printing.PDFForm doc, string field, DateTime? value, string format = "ddMMMyyyy", bool toUpper = false)
        {
            if (!value.HasValue)
            {
                return;
            }

            if (toUpper)
            {
                SetFormField(doc, field, value.Value.ToString(format).ToUpper());
            }
            else
            {
                SetFormField(doc, field, value.Value.ToString(format));
            }
        }

        public static void SetFormField(Printing.PDFForm doc, string field, string value)
        {
            if (!string.IsNullOrEmpty(field))
            {
                try
                {
                    doc.SetField(field, value);
                }
                catch (Exception ex)
                {
                    // Silently catch exceptions
                }
            }
        }

        public static string[] SplitString(string chopString, int length)
        {
            string top, bottom;
            string[] temp = new string[2];
            int dif = 0;

            if (string.IsNullOrEmpty(chopString) || chopString.Length == 0)
            {
                temp[0] = "";
                temp[1] = "";
            }
            else
            {
                if (chopString.Length > length)
                {
                    top = chopString.Substring(0, length);
                    temp = top.Split(' ');
                    dif = temp[temp.Length - 1].Length;

                    top = chopString.Substring(0, length - dif);
                    bottom = chopString.Substring(top.Length, chopString.Length - top.Length);

                    temp[0] = top;
                    temp[1] = bottom;
                }
                else
                {
                    temp[0] = chopString;
                    temp[1] = string.Empty;
                }
            }

            return temp;
        }
    }
}

File: ALODWebUtility/Printing/PrintingUtil.vb
Imports ALOD.Core.Domain.Modules.Lod
Imports ALOD.Core.Domain.Workflow
Imports ALOD.Core.Utils
Imports ALOD.Data.Services
Imports ALODWebUtility.Printing

Namespace PrintingUtil
    Public Module PrintingUtil

        Public Function CheckForPriorStatus(ByVal status As Short, ByVal refId As Integer) As Boolean
            Dim lod As LineOfDuty = LodService.GetById(refId)

            If (status = LodStatusCode.Complete) Then
                Dim ws As IList(Of WorkStatusTracking) = WorkFlowService.GetWorkStatusTracking(lod.Id, ModuleType.LOD)

                If (ws IsNot Nothing) Then
                    If (ws.Count > 1) Then
                        If (ws(1).WorkflowStatus.StatusCodeType.Id = LodStatusCode.AppointingAutorityReview) Then
                            Return True
                        Else
                            Return False
                        End If
                    End If
                End If
            End If

            Return False
        End Function

        Public Sub ClearFormField(ByVal doc As PDFForm, ByVal field As String)
            SetFormField(doc, field, String.Empty)
        End Sub

        Public Function GetFindingFormText(ByVal findingValue As Finding) As String
            Select Case findingValue
                Case Finding.In_Line_Of_Duty
                    Return "Line of Duty"
                Case Finding.Epts_Lod_Not_Applicable
                    Return "EPTS-LOD Not Applicable"
                Case Finding.Nlod_Due_To_Own_Misconduct
                    Return "Not ILOD-Due to Own Misconduct"
                Case Finding.Epts_Service_Aggravated
                    Return "EPTS-Service Aggravated"
                Case Finding.Nlod_Not_Due_To_OwnMisconduct
                    Return "Not ILOD-Not Due to Own Misconduct"
                Case Finding.Recommend_Formal_Investigation
                    Return "Formal Investigation"
                Case Else
                    Return String.Empty
            End Select
        End Function

        Public Function IsValidSignature(ByVal signature As SignatureEntry) As Boolean
            If (signature Is Nothing) Then
                Return False
            End If

            Return signature.IsSigned
        End Function

        Public Function RemoveNewLinesFromString(ByVal data As String) As String
            If (String.IsNullOrEmpty(data)) Then
                Return String.Empty
            End If

            Dim replacedString As String = data
            Dim replaceFinished As Boolean = False
            Dim len As Integer

            While (Not replaceFinished)
                len = replacedString.Length()
                replacedString = replacedString.Replace(Environment.NewLine, " ")

                If (len = replacedString.Length()) Then
                    replaceFinished = True
                End If
            End While

            Return replacedString
        End Function

        Public Sub SetCheckboxField(ByVal doc As PDFForm, ByVal field As String, ByVal value As Boolean?)
            If (Not value.HasValue OrElse value.Value = False) Then
                Exit Sub
            End If

            SetFormField(doc, field, "1")
        End Sub

        Public Sub SetDateTimeField(ByVal doc As PDFForm, ByVal field As String, ByVal value As DateTime?, Optional ByVal format As String = "ddMMMyyyy", Optional ByVal toUpper As Boolean = False)
            If (Not value.HasValue) Then
                Exit Sub
            End If

            If (toUpper) Then
                SetFormField(doc, field, value.Value.ToString(format).ToUpper())
            Else
                SetFormField(doc, field, value.Value.ToString(format))
            End If
        End Sub

        Public Sub SetFormField(ByVal doc As PDFForm, ByVal field As String, ByVal value As String)
            If (Not String.IsNullOrEmpty(field)) Then
                Try
                    doc.SetField(field, value)
                Catch ex As Exception

                End Try
            End If
        End Sub

        Function SplitString(ByVal chopString As String, ByVal length As Integer) As String()

            Dim top As String, bottom As String, temp(1) As String
            Dim dif As Integer = 0

            If (String.IsNullOrEmpty(chopString) OrElse chopString.Length = 0) Then
                temp(0) = ""
                temp(1) = ""
            Else
                If chopString.Length > length Then
                    top = chopString.Substring(0, length)
                    temp = Split(top, " ")
                    dif = temp(temp.GetUpperBound(0)).Length

                    top = chopString.Substring(0, length - dif)
                    bottom = chopString.Substring(top.Length, chopString.Length - top.Length)

                    temp(0) = top
                    temp(1) = bottom
                Else
                    temp(0) = chopString
                    temp(1) = String.Empty

                End If
            End If

            Return temp

        End Function

    End Module
End Namespace